v4-client-js/src/clients/native.ts

```diff
@@ -988,3 +1017,66 @@ export async function getMarketPrice(
     return wrappedError(e);
   }
 }
+
+export function getNobleAddress(): Promise<String> {
+  try {
+    if (globalThis.nobleWallet?.address === undefined) {
+      throw new UserError('wallet is not set. Call connectWallet() first');
+    }
+    return Promise.resolve(encodeJson(globalThis.nobleWallet.address));
+  } catch (error) {
+    return Promise.resolve(wrappedError(error));
+  }
+}
+
+export async function getNobleBalance(): Promise<String> {
+  try {
+    const client = await getNobleClient();
+    if (client === undefined) {
+      throw new UserError(
+        'client is not connected.',
+      );
+    }
+    if (globalThis.nobleWallet?.address === undefined) {
+      throw new UserError('wallet is not set. Call connectWallet() first');
+    }
+    const coin = await client.getAccountBalance(globalThis.nobleWallet.address, 'uusdc');
+    return encodeJson(coin);
+  } catch (error) {
+    return wrappedError(error);
+  }
+}
+
+export async function sendNobleIBC(squidPayload: string): Promise<String> {
+  try {
+    const client = await getNobleClient();
+    if (client === undefined) {
+      throw new UserError(
+        'client is not connected.',
+      );
+    }
+    if (globalThis.nobleWallet?.address === undefined) {
+      throw new UserError('wallet is not set. Call connectWallet() first');
+    }
+
+    const decode = (str: string): string => Buffer.from(str, 'base64').toString('binary');
+    const decoded = decode(squidPayload);
+
+    const json = JSON.parse(decoded);
+
+    const ibcMsg: EncodeObject = {
+      typeUrl: json.msgTypeUrl, // '/ibc.applications.transfer.v1.MsgTransfer',
+      value: json.msg,
+    };
+    const fee = await client.simulateTransaction([ibcMsg]);
+
+    // take out fee from amount before sweeping
+    ibcMsg.value.token.amount = (parseInt(ibcMsg.value.token.amount, 10) -
```

Comment:
Can we ensure ibcMsg.value.token.amount is positive?  Just to be safe.

Reply:
yes, can do.